powertools-metrics/src/main/java/software/amazon/lambda/powertools/metrics/internal/LambdaMetricsAspect.java

```diff
@@ -37,11 +37,18 @@ public class LambdaMetricsAspect {
     public static final String TRACE_ID_PROPERTY = "xray_trace_id";
     public static final String REQUEST_ID_PROPERTY = "function_request_id";
     private static final String SERVICE_DIMENSION = "Service";
+    private static final String FUNCTION_NAME_ENV_VAR = "POWERTOOLS_METRICS_FUNCTION_NAME";
 
     private String functionName(Metrics metrics, Context context) {
         if (!"".equals(metrics.functionName())) {
             return metrics.functionName();
         }
+
+        String envFunctionName = System.getenv(FUNCTION_NAME_ENV_VAR);
+        if (envFunctionName != null && !envFunctionName.isEmpty()) {
+            return envFunctionName;
+        }
+
```

Comment:
There's an annotation error from the CI at a line below, maybe we can address it even tho it's not part of the changes in this PR

![image](https://github.com/user-attachments/assets/62db035d-b606-416e-a519-eb46fab3618b)

Reply:
I will add a comment in this method explaining why it is needed.

Reply:
https://github.com/aws-powertools/powertools-lambda-java/pull/1863/commits/26a4b1dec83455305ccd0a5e154bc090c539c95e